parsing/src/parts.ts

```diff
@@ -109,55 +122,66 @@ function getItems(data: any[]): PartDataInterface {
                     stackSize: 500, //SS_HUGE
                     isFluid: false,
                     isFicsmas: true,
+                    energyGeneratedInMJ: 0
                 };                
             } else if (entry.ClassName === "Desc_Crystal_C") {
                 parts["Crystal"] = {
                     name: "Blue Power Slug",
                     stackSize: 50, //SS_SMALL
                     isFluid: false,
                     isFicsmas: false,
+                    energyGeneratedInMJ: 0
                 };                
             } else if (entry.ClassName === "Desc_Crystal_mk2_C") {
                 parts["Crystal_mk2"] = {
                     name: "Yellow Power Slug",
                     stackSize: 50, //SS_SMALL
                     isFluid: false,
                     isFicsmas: false,
+                    energyGeneratedInMJ: 0
                 };                
             } else if (entry.ClassName === "Desc_Crystal_mk3_C") {
                 parts["Crystal_mk3"] = {
                     name: "Purple Power Slug",
                     stackSize: 50, //SS_SMALL
                     isFluid: false,
                     isFicsmas: false,
+                    energyGeneratedInMJ: 0
                 };                
             } else if (entry.ClassName === "Desc_SAM_C") {
                 parts["SAM"] = {
                     name: "SAM",
                     stackSize: 100, //SS_MEDIUM
                     isFluid: false,
                     isFicsmas: false,
+                    energyGeneratedInMJ: 0
                 };                
             } else if (entry.ClassName === "Desc_CrystalShard_C") {
                 parts["CrystalShard"] = {
                     name: "Power Shard",
                     stackSize: 100, //SS_MEDIUM
                     isFluid: false,
                     isFicsmas: false,
+                    energyGeneratedInMJ: 0
                 }; 
             } else if (entry.ClassName === "BP_ItemDescriptorPortableMiner_C") {
                 parts["PortableMiner"] = {
                     name: "Portable Miner",
                     stackSize: 50, //SS_SMALL
                     isFluid: false,
                     isFicsmas: false,
+                    energyGeneratedInMJ: 0
                 }; 
             }
 
+            if (!entry.ClassName) return;
+
             // Ensures it's a recipe, we only care about items that are produced within a recipe.
             if (!entry.mProducedIn) return;
+            //if (!whitelist.some(part => entry.ClassName && entry.ClassName.includes(part)) && !entry.mProducedIn) return;
```

Comment:
Leftover code?

Reply:
coming back to it later - I felt like the scope of what I was trying to do was getting too big. I want to revisit after this merge.